src/renderer/renderer_types.zig

```diff
@@ -14,27 +15,31 @@ pub const MeshLod = struct {
     vertex_count: u32,
 };
 
-pub const BoundingBox = struct {
-    min: [3]f32,
-    max: [3]f32,
-};
+pub const SubMesh = struct {
+    lod_count: u32,
+    lods: [lod_count_max]MeshLod,
 
-pub const BoundingBoxCoordinates = struct {
-    center: [3]f32,
-    radius: f32,
+    bounding_box: BoundingBox,
 };
 
 pub const Mesh = struct {
     vertex_buffer: BufferHandle,
     index_buffer: BufferHandle,
-    num_lods: u32,
-    lods: [max_num_lods]MeshLod,
+
+    sub_mesh_count: u32,
```

Comment:
I think the std.BoundedArray (or List?) does what you want here. I mean this is fine (I'm sure I do it elsewhere as well) but it's worth knowing about since it means there's no risk of forgetting to keep the count up to date. This C-style is good for serialization and stuff though so sometimes it's valid I think :)

Reply:
Oh, I didn't know about std.BoundedArray! 
I use the C-style array + count in various places, so, if that's okay with you, I'll get familiar with the std.BoundedArray API after this PR and see where it makes sense.

Reply:
Yep ofc!